test/e2e/framework/kcp.go

```diff
@@ -395,6 +397,10 @@ func (c *kcpServer) DefaultConfig(t *testing.T) *rest.Config {
 	return rest.AddUserAgent(rest.CopyConfig(cfg), t.Name())
 }
 
+func (c *kcpServer) ClusterConfig(r *rest.Config) *rest.Config {
```

Comment:
I started a thread in Slack for this, but IMHO the round-tripper (with some extra logic) could handle the case where a cluster is provided statically, which would mean that constructing discovery clients would not need "raw" config and there would be no need to have any distinction between `DefaultConfig()` and `ClusterConfig()` - the fewer choices a dev needs to make, the fewer options for mistakes to be made, and the less complex everything ends up. I think we should take a stab at that.